ansible_base/jwt_consumer/hub/auth.py

```diff
@@ -51,25 +58,38 @@ def process_permissions(self):
                         team = Resource.objects.get(ansible_id=ansible_id).content_object
                     except Resource.DoesNotExist:
                         team = self.common_auth.get_or_create_resource('team', team_data)[1]
-                    teams.append(team)
 
                     groups.append(team.group)
 
+                    if role_name == 'Team Member':
+                        member_teams.append(team)
+
         self.common_auth.user.groups.set(groups)
 
+        '''
         # manage org membership ...
         org_pks = [org.pk for org in orgs]
         for org in Organization.objects.exclude(pk__in=org_pks).filter(users=self.common_auth.user):
             org.users.remove(self.common_auth.user)
         for org in orgs:
             org.users.add(self.common_auth.user)
+        '''
 
         # manage team membership ...
-        team_pks = [team.pk for team in teams]
-        for team in Team.objects.exclude(pk__in=team_pks).filter(users=self.common_auth.user):
-            team.users.remove(self.common_auth.user)
-        for team in teams:
-            team.users.add(self.common_auth.user)
+        member_team_pks = [team.pk for team in member_teams]
+        # delete all memberships not defined by this jwt ...
+        for assignment in RoleUserAssignment.objects.filter(
+            user=self.common_auth.user
+        ).exclude(object_id__in=member_team_pks):
+            assignment.delete()
+        # assign "local" membership for each team ...
+        roledef = RoleDefinition.objects.get(name='Galaxy Team Member')
+        for team in member_teams:
+            RoleUserAssignment.objects.get_or_create(
+                user=self.common_auth.user,
+                role_definition=roledef,
+                object_id=team.pk
+            )
```

Comment:
Can you replace the `get_or_create` with `roledef.give_permission(user, team)`? Sorry, I said signals before but it's not signals that DAB RBAC relies on. It relies on the code that is ran outright by `give_permission`